apps/portal/controllers/gadgets.jag

```diff
@@ -0,0 +1,99 @@
+ */
+
+var fresh = false;
+isDefault = false;
+var bgColor;
+var pageId;
+(function () {
+    var log = new Log();
+    var matcher = new URIMatcher(request.getRequestURI());
+    var uriElements;
+    var dashboardId;
+    var page;
+    var isStandAlone = true; //To check whether the request is for stand alone gadget/ pub-sub gadget
+    var i18n = require("i18n");
+    var gadgets;
+    i18n.init(request, utils.getLocaleResourcePath());
+
+    if (true) {
+        // To set the bgcolor according to client`s website template
+        bgColor = request.getParameter("bgcolor");
+        if (log.isDebugEnabled()) {
+            log.debug("Session id provided from the client is " + session.getId() + "Client has successfully connected.")
+        }
+        if (matcher.match('/gadgets/{id}/{pid}')) {
+            uriElements = matcher.elements();
+            dashboardId = uriElements.id;
+            pageId = uriElements.pid;
+            isStandAlone = false
+        }
+        else if (matcher.match('/gadgets/{id}/{pid}/{gid}')) {
+            // Get the dashboard id, page id and gadget id from url
+            uriElements = matcher.elements();
+            dashboardId = uriElements.id;
+            pageId = uriElements.pid;
+            var gadgetId = uriElements.gid;
+        }
+        else {
+            response.sendError(400, i18n.localize("bad.request"));
+        }
+
+        gadgets = require('/modules/gadgets.js');
+        utils.sandbox(context, function () {
+            dashboard = gadgets.getAsset(dashboardId);
+        });
+
+        // If the dashboard is not available, indicate error
+        if (!dashboard) {
+            response.sendError(404, i18n.localize("page.not.found"));
+            return;
+        }
+
+        // Check whether the there is page in the dashboard with given page id
+        page = gadgets.findPage(dashboard, pageId);
+        // If the page is not available indicate error
+        if (!page) {
+            response.sendError(404, i18n.localize("page.not.found"));
+            return;
+        }
+
+        // If it is a standalone gadget, check whether the required gadget exist
+        if (isStandAlone) {
+            // Check whether a gadget exist within the page
+            var isComponentExists = gadgets.isComponentExists(gadgetId, page);
+            // If there is no such gadget, send error.
+            if (!isComponentExists) {
+                response.sendError(404, i18n.localize("page.not.found"));
+                return;
+            }
+        }
+
+        // To make the user to view the the default view, since we are not authenticating by using user,
+        // but by using session id
+        isDefault = true;
+
+        // Based on the request (Whether it is stand alone or embedded) for the gadget, decide the template to render
+        isStandAlone ? include(utils.resolvePath('templates/gadget.jag')) :
+                include(utils.resolvePath('templates/pubsub-gadget.jag'));
```

Comment:
better be consistent with quotations for defining strings, use double quotes for outer strings and single quotes for inner strings.